### PR TITLE
Feature: extract dfget lib

### PR DESCRIPTION
**Pull request by wangforthinker**
### Ⅰ. Describe what this PR did
In the past, if request the file, dfdaemon should boot the dfget as a progress to register and pull file from peer node. To speed up the p2p download rate, we extract dfget lib as the inner progress in dfdaemon progress.

### Ⅱ. Does this pull request fix one issue?

NONE.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
Add test.

### CODE REVIEW

File: dfdaemon/config/config.go [hunk 1]
```diff
@@ -130,13 +131,12 @@ func (p *Properties) Validate() error {
...
 	}
 
-	if _, err := os.Stat(p.DFPath); err != nil && os.IsNotExist(err) {
-		return dferr.Newf(
-			constant.CodeExitDfgetNotFound,
-			"dfpath %s not found", p.DFPath,
-		)
-	}
-
+	// if _, err := os.Stat(p.DFPath); err != nil && os.IsNotExist(err) {
```
Comment (allencloud) [inline dfdaemon/config/config.go]: just remove them if you think the code is useless

File: cmd/dfdaemon/app/init.go [hunk 2]
```diff
@@ -58,14 +89,18 @@ func initDfdaemon(cfg config.Properties) error {
...
+	}
 
 	go cleanLocalRepo(cfg.DFRepo)
 
-	dfgetVersion, err := exec.Command(cfg.DFPath, "version").CombinedOutput()
-	if err != nil {
-		return errors.Wrap(err, "get dfget version")
-	}
-	logrus.Infof("use %s from %s", bytes.TrimSpace(dfgetVersion), cfg.DFPath)
+	// dfgetVersion, err := exec.Command(cfg.DFPath, "version").CombinedOutput()
```
Comment (allencloud) [inline cmd/dfdaemon/app/init.go]: just remove them if you think the code is useless

File: cmd/dfdaemon/app/init.go [hunk 3]
```diff
@@ -17,34 +17,65 @@
...
+	"github.com/dragonflyoss/Dragonfly/pkg/httputils"
+	"github.com/dragonflyoss/Dragonfly/pkg/netutils"
 	statutil "github.com/dragonflyoss/Dragonfly/pkg/stat"
+	"github.com/dragonflyoss/Dragonfly/pkg/stringutils"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+func adjustSupernodeList(nodes []string) []string {
```
Comment (allencloud) [inline cmd/dfdaemon/app/init.go]: please add comment for this function.

File: cmd/dfdaemon/app/init.go [hunk 4]
```diff
@@ -17,34 +17,65 @@
...
+		return append(nodes, nodes[0])
+	default:
+		util.Shuffle(nodesLen, func(i, j int) {
+			nodes[i], nodes[j] = nodes[j], nodes[i]
+		})
+		return append(nodes, nodes...)
+	}
+}
+
+func checkConnectSupernode(nodes []string) (localIP string) {
```
Comment (allencloud) [inline cmd/dfdaemon/app/init.go]: please add comment for this function.

File: cmd/dfdaemon/app/init.go (hunk 2 above)
Comment (wangforthinker) [reply cmd/dfdaemon/app/init.go]: done.

File: dfdaemon/config/config.go (hunk 1 above)
Comment (wangforthinker) [reply dfdaemon/config/config.go]: done.

File: cmd/dfdaemon/app/init.go (hunk 3 above)
Comment (wangforthinker) [reply cmd/dfdaemon/app/init.go]: done.

File: cmd/dfdaemon/app/init.go (hunk 4 above)
Comment (wangforthinker) [reply cmd/dfdaemon/app/init.go]: done.

File: cmd/dfdaemon/app/init.go (hunk 3 above)
Comment (allencloud) [reply cmd/dfdaemon/app/init.go]: Why to do this? Could you be more specific?

File: cmd/dfdaemon/app/init.go [hunk 5]
```diff
@@ -17,34 +17,67 @@
...
+	var (
+		e error
+	)
+	for _, n := range nodes {
+		ip, port := netutils.GetIPAndPortFromNode(n, dfgetcfg.DefaultSupernodePort)
+		if localIP, e = httputils.CheckConnect(ip, port, 1000); e == nil {
+			return localIP
+		}
+		logrus.Errorf("Connect to node:%s error: %v", n, e)
```
Comment (allencloud) [inline cmd/dfdaemon/app/init.go]: I am afraid that we had better use a `Warn` log level for this line of log.

File: dfdaemon/downloader/p2p/dfclient.go [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+func getTaskFileName(realTarget string, sign string) string {
+	return filepath.Base(realTarget) + "-" + sign
+}
+
+func getCid(localIP string, sign string) string {
+	return localIP + "-" + sign
+}
+
+func flattenHeader(header map[string][]string) []string {
```
Comment (allencloud) [inline dfdaemon/downloader/p2p/dfclient.go]: unit tests are needed.

File: cmd/dfdaemon/app/init.go (hunk 3 above)
Comment (wangforthinker) [reply cmd/dfdaemon/app/init.go]: This is just copy from https://github.com/dragonflyoss/Dragonfly/blob/master/dfget/core/core.go#L278.

File: cmd/dfdaemon/app/init.go (hunk 5 above)
Comment (wangforthinker) [reply cmd/dfdaemon/app/init.go]: done

File: dfdaemon/downloader/p2p/dfclient.go (hunk 6 above)
Comment (wangforthinker) [reply dfdaemon/downloader/p2p/dfclient.go]: added ut.

**Pull request closed without merging**